assets/script.js

```diff
@@ -123,10 +123,55 @@ function displayRecipes(url) {
       let priceEl = $("<li>").text(`Price per serving: ${pricePerServing}`);
       let timeEl = $("<li>").text(`Minutes to prepare meal: ${readyInMinutes}`);
       let caloriesEl = $("<li>").text(`Calories: ${calories}`);
-
-      recipeDiv.append(headerEl, recipeEl);
+      
+      // Modal button and properties added here
+        let buttonEl = $("<button>")
+        .addClass("btn btn-secondary modal-btns")
+        // add attribute as an identifier for modal button to use to get recipie info from API
+        .attr("id", mealID)
+        .text("Cooking instructions");
+
+      recipeDiv.append(headerEl, recipeEl, buttonEl);
       recipeEl.append(priceEl, timeEl, caloriesEl);
       recipesContainerDiv.append(recipeDiv);
     }
   });
 }
+
+
+
+
+// Code to control opening and closing of modal
+
+// Get the modal div from HTML
+var modal = $("#myModal");
+
+// Get the button that opens the modal
+var modalBtn = $(".modal-btn");
+
+// Get the <span> element that closes the modal
+var span = $(".close")[0];
+
+// When the user clicks the button, open the modal. This function is used
+// so appended buttons added by JS to document can be clicked on
+
+$(document).on("click",".modal-btns",openModal);
+
+function openModal (event){
+  event.preventDefault();
+  console.log("click test 1");
```

Comment:
I've tested this locally and it works great! Could remove the console log test now we know this handler is used?

Reply:
Ok yes, will remove.